include/boost/beast/core/impl/handler_ptr.ipp

```diff
@@ -52,8 +52,9 @@ handler_ptr(handler_ptr&& other)
 {
     if(other.t_)
     {
-        new(&h_) Handler(std::move(other.handler()));
-        other.handler().~Handler();
+        ::new((void*)std::addressof(h_))
```

Comment:
I would prefer to leave out C-style casts completely, because it is almost impossible to search for them

Reply:
Done.